Form/FormFlow.php

```diff
@@ -722,7 +754,16 @@ protected function applyDataFromSavedSteps() {
 	 * {@inheritDoc}
 	 */
 	public function createForm() {
-		return $this->createFormForStep($this->currentStepNumber);
+		$form = $this->createFormForStep($this->currentStepNumber);
+
+		if ($this->expired) {
+			if ($this->hasListeners(FormFlowEvents::FLOW_EXPIRED)) {
```

Comment:
Please put both conditions into one statement: `if ($this->expired && $this->hasListeners(FormFlowEvents::FLOW_EXPIRED)) {`